### PR TITLE
fix: TypeError exception when playing track with unnamed artists  #1991 

### PR DESCRIPTION
**Pull request by Kunal-Attri**
@kingosticks  Reopened PR for issue #1991  for previously closed PR #2011 

Review it, and give me a go ahead for actual implementation in main program

### CODE REVIEW

File: mopidy/core/history.py [hunk 1]
```diff
@@ -28,11 +28,20 @@ def _add_track(self, track):
...
+                    artist.name for artist in track.artists if artist.name
+                )
             )
         if track.name is not None:
             name_parts.append(track.name)
-        name = " - ".join(name_parts)
+        are_all_artist_none = True
+        for artist in track.artists:
+            if artist.name is not None:
+                are_all_artist_none = False
```
Comment (djmattyg007) [inline mopidy/core/history.py:40]: Insert a break statement after this line. There’s no reason to continue the loop once this is set to false.

Comment (Kunal-Attri) [reply mopidy/core/history.py:40]: Done

**Comment by djmattyg007** [comment]
Would you be able to squash your branch down to a single commit?

**Comment by Kunal-Attri** [comment]
Well, possibly, I have to fork the repo again and edit the whole thing and put in a single commit? Right?

**Comment by kingosticks** [comment]
No, you can do what is described at https://stackoverflow.com/questions/5667884/how-to-squash-commits-in-git-after-they-have-been-pushed

Personally I'd quite like to understand why my suggested alternate version failed some other tests.

**Comment by Kunal-Attri** [comment]
Now, what next is to be done?

**Comment by kingosticks** [comment]
In an ideal world you'd also add a changelog entry to go along with this but we can do that at some later point.

**Comment by kingosticks** [comment]
Thanks @Kunal-Attri

**Comment by Kunal-Attri** [comment]
Will remember about changelog entry in further contributions, this one was just my first contribution, so learning!
You're welcome @kingosticks

**Pull request merged**